handler.js

```diff
@@ -67,17 +78,43 @@ module.exports.start_build = (event, context, callback) => {
     } else if (pullRequest.state != "open") {
       callback("Pull request is not open");
     } else {
-      console.log("Cleared tests, this is a buildable event:", event)
+      console.log("Cleared tests, this is a buildable event:", event);
       response.pull_request = pullRequest;
       var head = pullRequest.head;
       var base = pullRequest.base;
       var repo = base.repo;
 
       var params = {
-        projectName: process.env.BUILD_PROJECT,
-        sourceVersion: 'pr/' + pullRequest.number
+        projectName: process.env.CB_BUILD_PROJECT
       };
 
+      if (!buildOptions.cbExternalBuildspec) {
+        params.sourceVersion = 'pr/' + pullRequest.number;
+        console.log("CodeBuild is using the source version (" + params.sourceVersion + ") (Use CB_EXTERNAL_BUILDSPEC=true parameter to change.)");
+      } else {
+        if (buildOptions.cbGitRepoEnv ) {
+          params.environmentVariablesOverride = [
+            {
+              name: buildOptions.cbGitRepoEnv,
+              type: "PLAINTEXT",
+              value: pullRequest.base.repo.clone_url
+            },
+          ]
+          if (buildOptions.cbGitRefEnv) {
+            params.environmentVariablesOverride = [
+              {
+                name: buildOptions.cbGitRefEnv,
+                type: "PLAINTEXT",
+                value: pullRequest.number + ""
+              },
+            ]
+          }
+        }
+        console.log("Not using the source version! Will likely use most-recently commited buildspec.yml instead. (Use CB_EXTERNAL_BUILDSPEC=true parameter to change.)");
```

Comment:
The message here I think can be more straightforward. "Using an external buildspec. Will not pass `sourceVersion` when starting the CodeBuild job. Set CB_EXTERNAL_BUILDSPEC=false to change."